cmd/kube-proxy/app/server.go

```diff
@@ -200,7 +200,7 @@ func newProxyServer(ctx context.Context, config *kubeproxyconfig.KubeProxyConfig
 	}
 
 	rawNodeIPs := getNodeIPs(ctx, s.Client, s.Hostname)
-	s.PrimaryIPFamily, s.NodeIPs = detectNodeIPs(ctx, rawNodeIPs, config.BindAddress)
+	s.PrimaryIPFamily, s.NodeIPs = detectNodeIPs(ctx, rawNodeIPs, config.NodeIPOverride, config.IPFamilyPolicy)
 
```

Comment:
at some point in `newProxyServer` you need to check if ipFamilyPolicy is RequireDualStack but !dualStackSupported.